arcgis-ios-sdk-samples/Geometry/List transformations/ListTransformationsViewController.swift

```diff
@@ -0,0 +1,178 @@
+        }
+        else {
+            datatumTransformations = AGSTransformationCatalog.transformationsBySuitability(withInputSpatialReference: inputSR, outputSpatialReference: outputSR)
+        }
+        
+        defaultTransformation = AGSTransformationCatalog.transformation(forInputSpatialReference: inputSR, outputSpatialReference: outputSR)
+        tableView.reloadData()
+    }
+    
+    func setPEDataURL() {
+        // find the PEDataRuntime folder from the documents directory
+        // added using iTunes
+        let path = NSSearchPathForDirectoriesInDomains(FileManager.SearchPathDirectory.documentDirectory, FileManager.SearchPathDomainMask.userDomainMask, true)
+        let subpaths = FileManager.default.subpaths(atPath: path[0])!
+        
+        // search for PEDataRuntime matches
+        let predicate = NSPredicate(format: "SELF MATCHES %@", "PEDataRuntime")
+        let peDataRuntimePaths = subpaths.filter({ (objc) -> Bool in
+            return predicate.evaluate(with: objc)
+        })
+
+        // use first matching path as path to PE data
+        if let documentPEDataRuntime = peDataRuntimePaths.first {
+            // found "PEDataRuntime" folder, create full url
+            let peDataURL = URL(fileURLWithPath: path[0]).appendingPathComponent(documentPEDataRuntime)
+            try? AGSTransformationCatalog.setProjectionEngineDirectory(peDataURL)
+        }
+        
+        setupTransformsList()
+    }
+
+    @IBAction func oderByMapExtentValueChanged(_ sender: Any) {
+        setupTransformsList()
+    }
+
+    //MARK: - TableView data source
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return datatumTransformations.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "DatumTransformCell", for: indexPath)
+
+        // get the selected transformation
+        let transformation = datatumTransformations[indexPath.row]
+
+        // if we're missing the grid files, detail which ones
+        var files = ""
+        if transformation.isMissingProjectionEngineFiles {
+            files = "Missing grid files"
+
+            if let gt = transformation as? AGSGeographicTransformation {
+                gt.steps.forEach { (step) in
+                    if step.isMissingProjectionEngineFiles {
+                        files.append(": " + step.projectionEngineFilenames.joined(separator: ","))
+                    }
+                }
+            }
+        }
+        
+        cell.textLabel?.text = transformation.name
+        cell.detailTextLabel?.text = files
+        
+        if let defaultTransform = defaultTransformation {
+            cell.isSelected = transformation.isEqual(to: defaultTransform)
```

Comment:
Please call `UITableView.selectRow(at:animated:scrollPosition:)` after setting `defaultTransformation`. Setting the selected state here only sets the display state of the cell. The table view resets this when the cell is displayed, so the default cell doesn't actually appear selected.

FYI: You could have used `transformation == defaultTransformation`. No need to unwrap `defaultTransformation` or explicitly call `isEqual(to:)`.

Reply:
Very nice, thanks.

Reply:
@mhdostal Yes. Make sure you reload the table view before selecting the row for the default transformation:

```swift
tableView.reloadData()
if let defaultTransformation = defaultTransformation,
    let row = datatumTransformations.index(of: defaultTransformation) {
    let indexPath = IndexPath(row: row, section: 0)
    tableView?.selectRow(at: indexPath, animated: false, scrollPosition: .middle)
}
```

Reply:
I've refactored this a bit; when the transformation array is reset/rebuilt, the project geometry is removed, as is the table highlighting.